com.dz4k.FruitCredits.json

```diff
@@ -0,0 +1,3397 @@
+        "runhaskell Setup install"
+      ],
+      "sources": [
+        {
+          "sha256": "22de71fde38b236d3e9168a832b5e1e75d1fb4f4028667bdf747b3b4c8c1529c",
+          "type": "archive",
+          "url": "https://hackage.haskell.org/package/character-ps-0.1/character-ps-0.1.tar.gz"
+        },
+        {
+          "dest-filename": "character-ps.cabal",
+          "sha256": "b38ed1c07ae49e7461e44ca1d00c9ca24d1dcb008424ccd919916f92fd48d9fe",
+          "type": "file",
+          "url": "https://hackage.haskell.org/package/character-ps-0.1/revision/0.cabal"
+        }
+      ]
+    },
+    {
+      "name": "os-string",
+      "buildsystem": "simple",
+      "builddir": false,
+      "build-commands": [
+        "echo '#! /usr/bin/env runhaskell' >Setup.txt",
+        "echo '> import Distribution.Simple' >>Setup.txt",
+        "echo '> main = defaultMain' >>Setup.txt",
+        "ln -s Setup.txt Setup.lhs || true",
+        "runhaskell Setup configure --prefix=/app",
+        "runhaskell Setup build",
+        "runhaskell Setup install"
+      ],
+      "sources": [
+        {
+          "sha256": "22fcc7d5fc66676b5dfc57b714d2caf93cce2d5a79d242168352f9eb0fe2f18a",
+          "type": "archive",
+          "url": "https://hackage.haskell.org/package/os-string-2.0.6/os-string-2.0.6.tar.gz"
+        },
+        {
+          "dest-filename": "os-string.cabal",
+          "sha256": "7699e7ae9bf74d056a62f384ceef8dfb2aa660f3f7c8016e9703f3b995e5e030",
+          "type": "file",
+          "url": "https://hackage.haskell.org/package/os-string-2.0.6/revision/0.cabal"
+        }
+      ]
+    },
+    {
+      "name": "hashable",
+      "buildsystem": "simple",
+      "builddir": false,
+      "build-commands": [
+        "echo '#! /usr/bin/env runhaskell' >Setup.txt",
+        "echo '> import Distribution.Simple' >>Setup.txt",
+        "echo '> main = defaultMain' >>Setup.txt",
+        "ln -s Setup.txt Setup.lhs || true",
+        "runhaskell Setup configure --prefix=/app",
+        "runhaskell Setup build",
+        "runhaskell Setup install"
+      ],
+      "sources": [
+        {
+          "sha256": "3baee4c9027a08830d148ec524cbc0471de645e1e8426d46780ef2758df0e8da",
+          "type": "archive",
+          "url": "https://hackage.haskell.org/package/hashable-1.4.7.0/hashable-1.4.7.0.tar.gz"
+        },
+        {
+          "dest-filename": "hashable.cabal",
+          "sha256": "573f3ab242f75465a0d67ce9d84202650a1606575e6dbd6d31ffcf4767a9a379",
+          "type": "file",
+          "url": "https://hackage.haskell.org/package/hashable-1.4.7.0/revision/0.cabal"
+        }
+      ]
+    },
+    {
+      "name": "data-fix",
+      "buildsystem": "simple",
+      "builddir": false,
+      "build-commands": [
+        "echo '#! /usr/bin/env runhaskell' >Setup.txt",
+        "echo '> import Distribution.Simple' >>Setup.txt",
+        "echo '> main = defaultMain' >>Setup.txt",
+        "ln -s Setup.txt Setup.lhs || true",
```

Comment:
why are these needed?

Reply:
The haskell modules in the manifest are based on the output of [cabal-flatpak](https://hackage.haskell.org/package/cabal-flatpak-0.0) and the article [How to Flatpak a Haskell App into Flathub](https://medium.com/@lettier/how-to-flatpak-a-haskell-app-into-flathub-86ef6d69e94d). Are there more up-to-date resources for bundling Haskell programs?

Reply:
No, we don't have haskell apps. This is probably the second or third one ever.

Reply:
and the part that is autogenerated should be in its own file to make it easier to maintain.

Reply:
Split the generated modules out to a new file.